package/src/cpumon.cpp

```diff
@@ -63,60 +63,60 @@ std::map<std::string, double> const cpumon::get_json_average_stats(
 
 // Collect related hardware information
 void const cpumon::get_hardware_info(nlohmann::json& hw_json) {
-  // Read some information from /proc/cpuinfo
-  std::ifstream cpuInfoFile{"/proc/cpuinfo"};
-  if (!cpuInfoFile.is_open()) {
-    std::cerr << "Failed to open /proc/cpuinfo" << std::endl;
+  // Define the command and run it
+  const std::vector<std::string> cmd = {"lscpu"};
+
+  auto cmd_result = prmon::cmd_pipe_output(cmd);
+
+  // If the command failed print an error and move on
+  if (cmd_result.first) {
+    std::cerr << "Failed to execute 'lscpu' to get CPU information (code "
+              << cmd_result.first << ")" << std::endl;
     return;
   }
 
-  // Metrics to read from the input
-  std::vector<std::string> metrics{"processor", "model name", "siblings",
-                                   "cpu cores"};
-  unsigned int nCPU = 0, nSiblings = 0, nCores = 0;
-
-  // Loop over the file
-  std::string line;
-  while (std::getline(cpuInfoFile, line)) {
+  // Map lscpu names to the desired ones in the JSON
+  const std::map<std::string, std::string> metricToName{
```

Comment:
Although it's not a big deal, make this an `unordered_map`... (just good practice)

Reply:
Done.